### PR TITLE
Schema - Fix boolean fields in various tables

### PR DESCRIPTION
**Pull request by monishdeb**
Overview
----------------------------------------
Followup PR of https://github.com/civicrm/civicrm-core/pull/22925 

Remaining items:

LocationType and OptionValue

### CODE REVIEW

**Comment by monishdeb** [comment]
@colemanw this is a partial patch, still half way to go. Will complete by tomorrow.

**Comment by eileenmcnaughton** [comment]
I think @colemanw's suggestion is a good idea - 

Current test fail is

api_v3_SyntaxConformanceTest::testCustomDataGet with data set #40 ('MailingComponent')
incorrect count returned from MailingComponent getcount
Failed asserting that 0 matches expected 1.

/home/jenkins/bknix-dfl/build/core-22954-8m5od/web/sites/all/modules/civicrm/Civi/Test/Api3TestTrait.php:199
/home/jenkins/bknix-dfl/build/core-22954-8m5od/web/sites/all/modules/civicrm/tests/phpunit/api/v3/SyntaxConformanceTest.php:923
/home/jenkins/bknix-dfl/build/core-22954-8m5od/web/sites/all/modules/civicrm/tests/phpunit/CiviTest/CiviUnitTestCase.php:262
/home/jenkins/bknix-dfl/extern/phpunit8/phpunit8.phar:671

**Comment by colemanw** [comment]
@monishdeb I just noticed that a search for `'is_active' => "DEFAULT 0` in the diff turns up several results.
We should change all those `is_active` fields to `DEFAULT 1` because that's generally the expected behavior; when you create a record you don't expect it to be disabled by default.

**Comment by monishdeb** [comment]
Hmm agree, and that will resolve most of the test failures in this patch. I am going to make that change of setting default to 1.

**Comment by monishdeb** [comment]
Pushed all the table changes, waiting for the test build report.

**Comment by colemanw** [comment]
retest this please

**Comment by colemanw** [comment]
retest this please

**Comment by colemanw** [comment]
@monishdeb the Jenkins console has the following error:
```
ERROR 1048 (23000) at line 4252: Column 'is_ssl' cannot be null
```

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by colemanw** [comment]
```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'is_default' cannot be null in /home/jenkins/bknix-dfl/build/core-22954-72vh/web/sites/all/modules/civicrm/Civi/Test.php on line 241
```

**Comment by colemanw** [comment]
@monishdeb could you please review/merge #23079 and then rebase to take the `FinancialAccount` bits out of this PR? I had to make some extra changes to a few of those fields.

**Comment by colemanw** [comment]
@monishdeb 
```
Parsing schema description /home/jenkins/bknix-dfl/build/core-22954-c2i3/web/sites/all/modules/civicrm/xml/schema/Schema.xml
Extracting database information
Extracting table information
Installing core22954c2i3test_svt6l schema

Fatal error: Uncaught PDOException: SQLSTATE[21S01]: Insert value list does not match column list: 1136 Column count doesn't match value count at row 6 in /home/jenkins/bknix-dfl/build/core-22954-c2i3/web/sites/all/modules/civicrm/Civi/Test.php on line 241
```

File: xml/schema/Financial/FinancialAccount.xml [hunk 1]
```diff
@@ -144,6 +144,8 @@
   <field>
     <name>is_reserved</name>
     <type>boolean</type>
+    <default>0</default>
+    <required>true</required>
```
Comment (colemanw) [inline xml/schema/Financial/FinancialAccount.xml:148]: @monishdeb these 2 additions are redundant (same thing exists 3 lines below)

**Comment by monishdeb** [comment]
Jenkin test this please

**Comment by colemanw** [comment]
@monishdeb 
```
PDOException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'is_default' cannot be null in /home/jenkins/bknix-dfl/build/core-22954-dd0s/web/sites/all/modules/civicrm/Civi/Test.php on line 241
```
Perhaps it would be good to reduce the size of this PR to just a few tables, merge-on-pass, and repeat for the next few tables, etc.

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by colemanw** [comment]
Thanks @monishdeb - our CI server appears to be down :(

**Comment by colemanw** [comment]
Retest this please

**Comment by colemanw** [comment]
@monishdeb only one failure now!
```
CRM_Core_FormTest::testOpeningForms with data set "Location Type" ('civicrm/admin/locationType?reset=1')
Undefined array key "vcard_name"
```

**Comment by monishdeb** [comment]
Jenkins test this please

**Comment by monishdeb** [comment]
phew finally .. @colemanw the test-build passed now :)

**Pull request merged**